osu.Game/Screens/Edit/Components/Timelines/Summary/Parts/MarkerPart.cs

```diff
@@ -16,17 +17,15 @@ namespace osu.Game.Screens.Edit.Components.Timelines.Summary.Parts
     /// </summary>
     internal class MarkerPart : TimelinePart
     {
-        private readonly Drawable marker;
-
-        public MarkerPart()
-        {
-            Add(marker = new MarkerVisualisation());
-        }
+        private Drawable marker;
 
         [BackgroundDependencyLoader]
         private void load(OsuColour colours)
         {
-            marker.Colour = colours.Red;
+            Add(marker = new MarkerVisualisation
+            {
+                Colour = colours.Red
+            });

```

Comment:
but.... why? This was the reason for creating these classes.

Reply:
not sure what you're referring to. I just moved the add to the load method so the object initialiser could be used for the colour, nothing more.

Reply:
I'm referring to exactly that. The colour was the primary purpose of these `*Visualisation` classes. Unless you want to go back and remove every single such class from BreakPart, BookmarkPart, and ControlPointPart.

Reply:
i'm still not sure i follow sorry.

Reply:
https://github.com/peppy/osu/blob/d5ed218488aac139909012bb0343a69edf357421/osu.Game/Screens/Edit/Components/Timelines/Summary/Parts/BookmarkPart.cs#L31 https://github.com/peppy/osu/blob/d5ed218488aac139909012bb0343a69edf357421/osu.Game/Screens/Edit/Components/Timelines/Summary/Parts/BreakPart.cs#L32 https://github.com/peppy/osu/blob/d5ed218488aac139909012bb0343a69edf357421/osu.Game/Screens/Edit/Components/Timelines/Summary/Parts/ControlPointPart.cs#L59 https://github.com/peppy/osu/blob/d5ed218488aac139909012bb0343a69edf357421/osu.Game/Screens/Edit/Components/Timelines/Summary/Parts/ControlPointPart.cs#L48

are you also going to remove these?

Reply:
those ones can't be merged can they? the construction is in an inherited class..

Reply:
You can do almost the exact same thing you did here. But I don't like this. The marker should control its own colour, it's a private object.